api/src/controllers/team.js

```diff
@@ -71,13 +71,19 @@ async function storeHandle(req, res, next) {
 }
 
 /**
- * Gets the team document from the auth middleware
- * @param {*} req request object, containing the team object 
- * @param {*} res response object, the team object 
- * @returns 200: return the team passed by the auth middleware 
+ * Gets the team info
+ * @param {*} req request object contains the teamId decoded in auth middleware
+ * @param {*} res response object, the team related info 
+ * @returns 200: the team related info  
  */
-function getTeam(req, res) {
-  return res.status(200).send(req.team);
+function getTeam(req, res, next) {
+  Team.findById(req.team._id).select('_id teamName orgName email')
+  .then((foundTeam) => {
+    if (foundTeam) {
+      return res.status(200).send(foundTeam);
+    }
+  })
+  .catch((err) => next(fillErrorObject(500, 'Server error', [err.errors])));
```

Comment:
Because the get team logic has been moved into here, we might not need line 26 in middleware/team.js - because validating the teamId requires getting the team document anyway, getTeam and the middleware would be doing the same work. Up to you whether you want to change it or leave it as is.

Reply:
This get team endpoint requires the team_id, which is decoded from jwt in the previous auth middleware, not the team middleware. As this endpoint doesn't request any parameters sent from the frontend, only the accessToken is needed. 

While the validateTeamId middleware validates the teamId in the parameters sent from the frontend. 

They are doing different things though but will be a good idea to combine them together into only one middleware function, may be we can raise a ticket to investigate the way we can do that and refractor it.

Reply:
Yeah, agree with combining them. Will put that on the board.